internal/worker/task.go

```diff
@@ -82,6 +83,35 @@ func (worker *Worker) startTask(
 	worker.logger.Infof("started task %d", agentAwareTask.TaskId)
 }
 
+func (worker *Worker) getInstance(ctx context.Context, isolation *api.Isolation) (abstract.Instance, error) {
+	if standbyInstance := worker.standbyInstance; standbyInstance != nil {
+		// Relinquish our ownership of the standby instance since
+		// we'll either return it to the task or terminate it
+		worker.standbyInstance = nil
+
+		// Return the standby instance if matches the isolation required by the task
+		if proto.Equal(worker.standbyIsolation, isolation) {
+			worker.logger.Debugf("standby instance matches the task's isolation configuration, " +
+				"yielding it to the task")
+
+			return standbyInstance, nil
+		}
+
+		// Otherwise terminate the standby instance to simplify the resource management
+		worker.logger.Debugf("standby instance does not match the task's isolation configuration, " +
+			"terminating it")
+
+		if err := worker.standbyInstance.Close(ctx); err != nil {
```

Comment:
`worker.standbyInstance` is `nil` here already

Reply:
Good catch, fixed in https://github.com/cirruslabs/cirrus-cli/pull/700/commits/364b6f2a3906d8fa21733760606a6a522a95e14a.